### PR TITLE
Catalog of my things Version 1.0.0 

### PR DESCRIPTION
**Pull request by jlvFlores**
## Hello future code reviewer 👋
For this project we complied with the following requirements as our team comprised of 2 members that would me Cisco ( @Cisco-the-wayword ) and me. ( @jlvFlores )

#### Logic
- Each class from the [UML class diagram](./images/catalog_of_my_things.png) should be created in a separate .rb file. **(Not including movies and sources)**
- All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes should be implemented.
- All properties visible in the diagram should be defined and set up in the constructor method. _Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created._
- All methods visible in the diagram should be implemented:
    - add_item method (in all classes that have that method)
        - should take an instance of the Item class as an input.
        - should add the input item to the collection of items.
        - should add self as a property of the item object (by using the correct setter from the item object).
    - can_be_archived?() in the Item class
        - should return true if published_date is older than 10 years.
        - otherwise, it should return false.
    - move_to_archive() in the Item class
        - should reuse can_be_archived?() method.
        - should change the archived property to true if the result of the can_be_archived?() method is true.
        - should do nothing if the result of the can_be_archived?() method is false.
    - can_be_archived?() in the Book class
        - should override the method from the parent class.
        - should return true if parent's method returns true OR if cover_state equals to "bad".
        - otherwise, it should return false.
    - can_be_archived?() in the MusicAlbum class
        - should override the method from the parent class.
        - should return true if parent's method returns true AND if on_spotify equals true.
        - otherwise, it should return false.
    - can_be_archived?() in the Game class
        - should override the method from the parent class.
        - should return true if parent's method returns true AND if last_played_at is older than 2 years.
        - otherwise, it should return false.
- Add unit tests for all implemented methods.

#### User interface
- Create a main.rb file that will serve as your console app entry-point.
- Your console app, at the start, should:
    - Present the user with a list of options to perform.
    - Let users choose an option.
    - If needed, ask for parameters for the option.
    - Have a way to quit the app.
- The following options should be available:
    - List all books
    - List all music albums
    - List of games
    - List all genres (e.g 'Comedy', 'Thriller')
    - List all labels (e.g. 'Gift', 'New')
    - List all authors (e.g. 'Stephen King')
    - Add a book
    - Add a music album
    - Add a game
 - All data should be preserved by saving collections in .json files.

### Database
- Created a schema.sql file with tables that will are analogical to the structure of the classes in the app. 

### Project documentation
A [video](https://drive.google.com/file/d/1O0AY-0lq8AWu-bSShIZdy_u0z9KYdzfQ/view?usp=sharing) presenting the features of the project built. You can also find the video on the README.md file
The video includes:

- A description of the project.
- A demo of the project features.
- Highlight some interesting piece of code or something you built that you are very proud of.
- All should appear in the video and talk about the project.

### Workload distribution
We created a [Kanban board](https://github.com/jlvFlores/catalog-of-my-things/projects/1) in which you can see how we divided the task. 

We worked in this way:
- The group tasks (e.g., set up repo) were divided among all members or completed as a team (pair programming).
- The other tasks related to the project will be developed individually. 
  - Every student was given their own tasks.
- All tasks were based on the cards from the Kanban board.

### CODE REVIEW

File: spec/game_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+describe Game do
+  let(:author) { Author.new('Hidetaka', 'Miyazaki') }
+  let(:today) { Date.today.to_s }
+  let(:multiplayer) { true }
+  let(:over_decade) { Date.parse(today).prev_year(11).to_s }
+
+  context 'when receiving valid parameters' do
+    let(:game) { Game.new(author, today, multiplayer, over_decade) }
+
```
Comment (codepantha) [inline spec/game_spec.rb:13]: - [ ] One of the project requirements is that all public methods in all classes should have unit tests. Please write unit tests for the `to_hash` method in your `Game` class 👍

File: spec/genre_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    it 'returns a valid Genre object' do
+      expect(genre).to be_instance_of(Genre)
+    end
+
+    it 'add_item method updates the items instance variable' do
+      genre.add_item(item)
+      expect(genre.items).to eq([item])
+    end
+  end
```
Comment (codepantha) [inline spec/genre_spec.rb:16]: - [ ] One of the project requirements is that all public methods in all classes should have unit tests. Please write unit tests for the `to_hash` method in your `Genre` class 👍

File: spec/music_album_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+describe MusicAlbum do
+  let(:genre) { Genre.new('Rock') }
+
+  context 'when receiving valid parameters' do
+    let(:music_album) { MusicAlbum.new(genre, '2012-07-10', true) }
+
+    it 'returns a valid MusicAlbum object' do
+      expect(music_album).to be_instance_of(MusicAlbum)
+    end
+  end
```
Comment (codepantha) [inline spec/music_album_spec.rb:14]: - [ ] One of the project requirements is that all public methods in all classes should have unit tests. Please write unit tests for the `to_hash` method in your `MusicAlbum` class 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## HIGHLIGHTS 🎯 
✔️ Descriptive PR
✔️ Wrote unit tests 📌 
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lizdev-05 — APPROVED** [review]
Hi @jlvFlores @codepantha  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

 :heavy_check_mark: Professional Readme file
:heavy_check_mark: All your tests pass +1
:heavy_check_mark: No linter error

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**